rollout/controller_test.go

```diff
@@ -792,6 +792,12 @@ func (f *fixture) expectPatchServiceAction(s *corev1.Service, newLabel string) i
 	return len
 }
 
+func (f *fixture) expectGetReplicaSetAction(r *appsv1.ReplicaSet) int { //nolint:unused
+	len := len(f.kubeactions)
+	f.kubeactions = append(f.kubeactions, core.NewGetAction(schema.GroupVersionResource{Resource: "replicasets"}, r.Namespace, r.Name))
+	return len
+}
```

Comment:
we should probably add several test cases to validate the new `updateReplicaSetFallbackToPatch` method

Reply:
There are two test that cover most the happy paths to lock in the behavior on the two conflict points [here](https://github.com/argoproj/argo-rollouts/pull/3559/files#diff-5cdac8975ba1fb18da441a33b6cee881a5d8d6006bf5b74155080c20bed9c02bR2150-R2245). I don't think it's worth going down a lot of the error paths like incorrect json marshal's etc.

Reply:
codecovs percentages don't always make sense to me, here is the coverage of that function https://app.codecov.io/gh/argoproj/argo-rollouts/pull/3559?src=pr&el=tree&filepath=rollout%2Fcontroller.go&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=argoproj#6b8a01c38fada35656503847a6f4a00d-R951

Reply:
It seems that the coverage is comparing the wrong base from the link above.